eth_utils/address.py

```diff
@@ -119,65 +118,72 @@ def _normalize_32byte_address(address):
         raise ValueError("Invalid address.  Must be 32 byte value")
 
 
-@coerce_args_to_text
-@coerce_return_to_text
-def to_normalized_address(address):
+def to_normalized_address(primitive=None, hexstr=None, text=None):
     """
     Converts an address to it's normalized hexidecimal representation.
     """
-    if is_hex_address(address):
+    address = to_text(primitive, hexstr, text)
+    if is_hex_address(text=address):
         return _normalize_hex_address(address)
-    elif is_binary_address(address):
+    elif is_binary_address(text=address):
         return _normalize_binary_address(address)
-    elif is_32byte_address(address):
+    elif is_32byte_address(text=address):
         return _normalize_32byte_address(address)
 
     raise ValueError("Unknown address format {0}".format(address))
 
 
-def is_normalized_address(value):
+def is_normalized_address(primitive=None, hexstr=None, text=None):
```

Comment:
I think the original version of this function will work if all of the other changes are made/reverted.